invenio_users_resources/entity_resolvers.py

```diff
@@ -12,34 +12,50 @@
 from types import SimpleNamespace
 
 from flask_principal import UserNeed
+from invenio_access.permissions import system_process, system_user_id
 from invenio_accounts.models import User
 from invenio_records_resources.references.entity_resolvers import (
     EntityProxy,
     EntityResolver,
 )
+from sqlalchemy.exc import NoResultFound
 
 from .proxies import current_users_service
-from .services.schemas import UserGhostSchema
+from .services.schemas import SystemUserSchema, UserGhostSchema
 from .services.users.config import UsersServiceConfig
 
 
 class UserProxy(EntityProxy):
     """Resolver proxy for a User entity."""
 
     def _resolve(self):
-        """Resolve the User from the proxy's reference dict."""
-        user_id = int(self._parse_ref_dict_id())
-        return User.query.get(user_id)
+        """Resolve the User from the proxy's reference dict, or system_identity."""
+        user_id = self._parse_ref_dict_id()
+        if user_id == system_user_id:  # system_user_id is a string: "system"
+            return self.system_record()
+        else:
+            try:
+                return User.query.get(int(user_id))
+            except NoResultFound:
+                return self.ghost_record(user_id)
```

Comment:
Just for consistency reasons: do we want to dump the `user_id` as string or int? Or does this not matter, as it is a deleted user anyway?